src/scheduled_tasks.rs

```diff
@@ -48,9 +52,10 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
     update_banned_when_expired(&mut conn_1);
   });
 
-  // Update hot ranks every 5 minutes
+  // Update hot and controversy ranks every 5 minutes
   scheduler.every(CTimeUnits::minutes(5)).run(move || {
     update_hot_ranks(&mut conn_2, true);
+    update_controversy_ranks(&mut conn_2, true);
```

Comment:
Here's the main issue with this PR, that should get you to rethink it. The **only reason** hot ranks need to be recalculated, is because its a decaying time function. I forget the term in SQL, but its essentially based on an unstable function.

A controversial sort, which works only on upvotes, downvotes, and score, is not that. It can be done entirely with vote triggers.

Just like the `post_aggregates.score` column, you need to update the vote triggers to run whatever logic necessary to update that controversial_rank column.